### PR TITLE
fix: updating ViewInspector version

### PR DESCRIPTION
**Pull request by benmckillop-dd**
# fix: updating ViewInspector version

_Thank you for your contribution to the project._

ViewInspector broke for Xcode 16 and requires an update to v0.10.1 to work locally for Xcode 16.

Some context here on why it broke: 
- https://github.com/nalexn/ViewInspector/issues/327
- https://forums.swift.org/t/any-info-on-the-typeeraser-attribute/37826/2

This is blocking us a bit so would be good to get in asap.

# Checklist

## Before raising your pull request:
- [x] Ran the app locally ensuring it builds 
- [x] Ran the tests locally ensuring they pass on Build
- [x] Pull request has a clear title with ticket ID and a short description about the feature or update
      i.e. _DCMAW-222: Added ReadID SDK to iOS app_
- [x] Created a `draft` pull request if it is not yet ready for review

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Jira
- [x] Reviewed your own code to ensure you are following the style guidelines
- [x] Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.
- [x] Written Unit and Integration tests if needed

- [x] Met all accessibility requirements?
    - [x] Checked dynamic type sizes are applied
    - [x] Checked VoiceOver can navigate your new code
    - [x] Checked a user can navigate only using a keyboard around your new code 

## Before merging your pull request:
- [ ] Ensure that the code coverage and SonarCloud checks have passed
- [ ] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [ ] Ran the app to ensure that no regressions have been caused by changes during code review.
- [ ] Targeted the correct branch; `develop`, `release` or `main`

### CODE REVIEW

**Review by shukriahmed-dd — APPROVED** [review]

File: Package.swift [hunk 1]
```diff
@@ -10,9 +10,8 @@ let package = Package(
         .library(name: "GDSCommon", targets: ["GDSCommon"])
     ],
     dependencies: [
-        .package(
-            url: "https://github.com/nalexn/ViewInspector",
-            .upToNextMajor(from: "0.9.7"))
+        .package(url: "https://github.com/nalexn/ViewInspector",
+                 .upToNextMajor(from: .init(0, 10, 1)))
```
Comment (robinphillips) [inline Package.swift:14]: should we be pinning this to a specific git commit hash?

Comment (robinphillips) [reply Package.swift:14]: Sure, I guess it's maybe a bigger discussion and not worth deviating for this one on its own. Reason for doing it is that it helps protect against a situation where a dependency is taken over by a malicious actor.

**Review by robinphillips — APPROVED** [review]

File: Package.swift (hunk 1 above)
Comment (benmckillop-dd) [reply Package.swift:14]: Yeah I think maybe a bigger conversation around our packages, so far we have been following the above :)

**Review by dylanjames-dd — APPROVED** [review]
🙌

**Pull request merged**